common/src/main/java/org/jboss/eap/qe/ts/common/docker/Docker.java

```diff
@@ -0,0 +1,188 @@
+package org.jboss.eap.qe.ts.common.docker;
+
+import org.fusesource.jansi.Ansi;
+import org.junit.rules.ExternalResource;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Intended to be used as a JUnit {@link org.junit.ClassRule ClassRule}.
+ *
+ * Example of usage:
+ *
+ * import org.jboss.arquillian.container.test.api.RunAsClient;
+ * import org.jboss.arquillian.junit.Arquillian;
+ * import org.jboss.arquillian.junit.InSequence;
+ * import org.jboss.eap.qe.ts.common.docker.Docker;
+ * import org.jboss.eap.qe.ts.common.docker.DockerContainers;
+ * import org.junit.ClassRule;
+ * import org.junit.Test;
+ * import org.junit.runner.RunWith;
+ *
+ * @RunAsClient
+ * @RunWith(Arquillian.class)
+ * public class TestDocker {
+ *
+ * // Docker container will be started in @BeforeClass phase - see list and current configuration of current {@link DockerContainers}
+ * @ClassRule
+ * public static Docker jaegerContainer=DockerContainers.jaeger();
+ *
+ *
+ * @Test
+ * public void test()  {
+ *  // test something with docker container...
+ * }
+ * }
+ */
+public class Docker extends ExternalResource {
+    private final String uuid;
+    private final String name;
+    private final String image;
+    private final List<String> ports = new ArrayList<>();
+    private final Map<String, String> environmentVariables = new HashMap<>();
+    private final List<String> commandArguments = new ArrayList<>();
+    private String awaitedLogLine;
+    private long waitTimeoutInMillis = 600_000; // 10 minutes
+
+    private final ExecutorService outputPrinter = Executors.newSingleThreadExecutor();
+
+    public Docker(String name, String image) {
+        this.uuid = UUID.randomUUID().toString();
+        this.name = name;
+        this.image = image;
+    }
+
+    public Docker waitForLogLine(String logWait) {
```

Comment:
These methods are in fact setters, yet their name doesn't reflect this fact. From the name I would guess that calling this method causes some kind of wait. Also, JavaDoc is missing...

Reply:
@honza-kasik thanks for the review.

Yes, this is copy-pasty from TT QE test suite. I didn't make any modification into API or logic.

In case of Builder pattern having `set...()` methods is not must have. It's more about to have descriptive names. For example `StringBuilder.append()`. 

One argument why to to keep current method names is that it will make migration easier and we will not have to rewrite it but looking at TT QE test suite this is basically used just in `DockerContainers` class so it's not big effort.

I'll change the names of methods to better reflect what is set. I would like to avoid writing JavaDoc for those methods as those should be self-explanatory from method names.

Reply:
Thanks for considering this! Although this look like a `Builder` pattern it is IMHO not. There is no separation between the builder and product. So these are in my opinion setters and not methods of builder.

Reply:
@honza-kasik Don't worry I've changed the method names expect this one :-) Could you take a look, please?